h2o-admissibleml/src/main/java/hex/Infogram/InfogramUtils.java

```diff
@@ -0,0 +1,328 @@
+        break;
+      case drf:
+        paramsSchema = new DRFV3.DRFParametersV3();
+        break;
+      case deeplearning:
+        paramsSchema = new DeepLearningV3.DeepLearningParametersV3();
+        break;
+      default:
+        throw new UnsupportedOperationException("Unknown algo: " + algoName);
+    }
+    Model.Parameters[] modelParams = new Model.Parameters[numModels];
+    for (int index = 0; index < numModels; index++) {
+      modelParams[index] = (Model.Parameters) paramsSchema.fillFromImpl(infoParams).createAndFillImpl();
+      modelParams[index]._ignored_columns = null; // training frame contains only needed columns
+      modelParams[index]._train = trainingFrames[index]._key;
+    }
+    return modelParams;
+  }
+
+  public static ModelBuilder[] buildModelBuilders(Model.Parameters[] modelParams) {
+    int numModel = modelParams.length;
+    ModelBuilder[] modelBuilders = new ModelBuilder[numModel];
+    for (int index = 0; index < numModel; index++)
+      modelBuilders[index] = ModelBuilder.make(modelParams[index]);
+    return modelBuilders;
+  }
+
+  /***
+   * Calculate the cmi for each predictor.  Refer to https://h2oai.atlassian.net/browse/PUBDEV-8075 section I step 2 
+   * for core infogram, or section II step 3 for fair infogram 
+   * 
+   * @param builders
+   * @param trainingFrames
+   * @param cmi
+   * @param startIndex
+   * @param numModels
+   * @param response
+   * @param generatedFrameKeys
+   */
+  public static void generateInfoGrams(ModelBuilder[] builders, Frame[] trainingFrames, double[] cmi, int startIndex,
+                                       int numModels, String response, List<Key<Frame>> generatedFrameKeys) {
+    for (int index = 0; index < numModels; index++) {
+      Model oneModel = builders[index].get();                   // extract model
+      Frame prediction = oneModel.score(trainingFrames[index]); // generate prediction
+      prediction.add(response, trainingFrames[index].vec(response));
+      Scope.track_generic(oneModel);
+      generatedFrameKeys.add(prediction._key);
+      cmi[index+startIndex] = new hex.Infogram.EstimateCMI(prediction).doAll(prediction)._meanCMI; // calculate raw CMI
+    }
+  }
+  
+  public static void removeFromDKV(List<Key<Frame>> generatedFrameKeys) {
+    for (Key<Frame> oneFrameKey : generatedFrameKeys)
+        DKV.remove(oneFrameKey);
+  }
+
+  /**
+   * If user choose GLM to derive the infogram metrics, need to reset the distribution to family.
+   * @param glmParam
+   * @param infoGramParam
+   */
+  public static void setGLMFamilyParams(GLMParameters glmParam, InfogramParameters infoGramParam) {
+    switch (infoGramParam._distribution) {
+      case bernoulli:
+        glmParam._family = binomial;
+        break;
+      case multinomial:
+        glmParam._family = multinomial;
+        break;
+      case ordinal:
+        glmParam._family = ordinal;
+        break;
+      default:  // no change.
+        break;
+    }
+  }
```

Comment:
It would be nice to have this kind of functionality somewhere outside of this module.

In Stacked Ensemble, we try to inherit distribution/family from base models so we do something similar - it might be worth refactoring it and having one reusable logic so it is easier to maintain/add new distributions/families...
https://github.com/h2oai/h2o-3/blob/4263969349fdd4106e34ea5a49bf83fcfca03ef8/h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java#L358-L530

To be clear I'm not saying you should do it in this PR I just wanted to inform you that this kind of conversion (Family -> Distribution and vice versa) is used somewhere else as well.

Reply:
Well, since I am using it, I may as well be the one to fix this.  Stay tuned.  I want to do it.

Reply:
Done.  Please check